osf/utils/migrations.py

```diff
@@ -163,12 +198,334 @@ def ensure_schemas(*args):
+            )
+            for question in page['questions']:
+                format_question(state, rs, question)
+
+
+def unmap_schemablocks(*args):
+    state = args[0]
+    RegistrationSchemaBlock = state.get_model('osf', 'registrationschemablock')
+    RegistrationSchemaBlock.objects.all().delete()
+
+
+# For registration_responses validation
+def get_multiple_choice_options(registration_schema, question, required_fields):
+    """
+    Returns a dictionary with an 'enum' key, and a value as
+    an array with the possible multiple choice answers for a given question.
+    Schema blocks are linked by schema_block_group_keys, so fetches multiple choice options
+    with the same schema_block_group_key as the given question
+    :question SchemaBlock with an registration_response_key
+    :required_fields boolean - do we want to enforce that required fields are present
+    """
+    options = registration_schema.schema_blocks.filter(
+        schema_block_group_key=question.schema_block_group_key,
+        block_type='select-input-option'
+    ).values_list('display_text', flat=True)
+
+    options = list(options)
+    # required is True if we want to both enforce required_fields
+    # and the question in particular is required.
+    required = required_fields and question.required
+    if not required and '' not in options:
+        options.append('')
+
+    return {
+        'enum': options
+    }
+
+# For registration_responses validation
+def get_jsonschema_type(block_type):
+    """
+    For a given schema block type, returns the corresponding
+    jsonschema type
+    :params block_type: string, SchemaBlock block_type
+    :return string
+    """
+    if block_type == 'file-input':
+        return 'array'
+    elif block_type == 'multi-select-input':
+        return 'array'
+    else:
+        return 'string'
+
+# For registration_responses validation
+def format_question_validation(registration_schema, question, required_fields):
+    """
+    Returns json for validating an individual question
+    :params question SchemaBlock
+    """
+    if question.block_type == 'single-select-input':
+        property = get_multiple_choice_options(registration_schema, question, required_fields)
+    elif question.block_type == 'multi-select-input':
+        property = {
+            'items': get_multiple_choice_options(registration_schema, question, required_fields)
+        }
+    elif question.block_type == 'file-input':
+        property = {
+            'items': FILE_UPLOAD_SCHEMA
+        }
+    else:
+        property = {}
+
+    property['type'] = get_jsonschema_type(question.block_type)
+    # Stashing the question title on the jsonschema's description field
+    property['description'] = registration_schema.schema_blocks.get(
+        schema_block_group_key=question.schema_block_group_key,
+        block_type='question-label'
+    ).display_text
+    return property
+
+# For registration_responses validation
```

Comment:
note: check that legacy workflow works with new validation turned on

Reply:
Did you verify this, and is there a note to QA to also verify this?